src/hotspot/make/bsd/makefiles/gcc.make

```diff
@@ -317,6 +319,14 @@ ifeq ($(USE_CLANG), true)
   ifeq ($(shell expr $(CC_VER_MAJOR) = 4 \& $(CC_VER_MINOR) = 2), 1)
     OPT_CFLAGS/loopTransform.o += $(OPT_CFLAGS/NOOPT)
     OPT_CFLAGS/unsafe.o += -O1
+    OPT_CFLAGS/metaspaceShared.o += $(OPT_CFLAGS/NOOPT)
+  endif
+
+  # We suspect that this bug was not addressed in Clang 6. Deopt unsafe.o
+  ifeq ($(shell expr $(CC_VER_MAJOR) = 6), 1)
+    OPT_CFLAGS/loopTransform.o += $(OPT_CFLAGS/NOOPT)
+    OPT_CFLAGS/unsafe.o += -O1
+    OPT_CFLAGS/metaspaceShared.o += $(OPT_CFLAGS/NOOPT)
```

Comment:
Could you leave paper trail here why metaspaceShared.cpp also needs special handle?
Further, please switch this line with 328.